backend/src/test/java/com/pickpick/message/application/ReminderServiceTest.java

```diff
@@ -61,293 +67,361 @@ class ReminderServiceTest {
+
+                assertThat(foundReminders).hasSize(count);
+            }
+        }
+
+        @DisplayName("요청 파라미터에 리마인더 id만 있다면")
+        @Nested
+        class onlyReminderIdInParams {
+
+            Reminder target = bomReminders.get(5);
+            ReminderFindRequest request = ReminderFindRequestFactory.onlyReminderId(target.getId());
+
+            @DisplayName("해당 리마인더보다 미래 시간의 리마인더를 기본 20개까지 조회한다")
+            @Test
+            void findFutureReminders() {
+                List<ReminderResponse> foundReminders = reminderService.find(request, bom.getId()).getReminders();
+                List<Long> expectedIds = extractFutureFromTargetOrderedByRemindDateIds(bomReminders, target, 20);
+
+                assertThat(foundReminders).extracting("id").containsExactlyElementsOf(expectedIds);
+            }
+        }
+
+        @DisplayName("요청 파라미터에 리마인더 id와 count가 있다면")
+        @Nested
+        class reminderIdAndCountInParams {
+
+            Reminder target = bomReminders.get(5);
+            int count = 3;
+
+            ReminderFindRequest request = ReminderFindRequestFactory.reminderIdAndCount(target.getId(), count);
+
+            @DisplayName("해당 리마인더보다 미래 시간의 리마인더를 count개 조회한다")
+            @Test
+            void findCountRemindersSetRemindedAfterTarget() {
+                List<ReminderResponse> foundReminders = reminderService.find(request, bom.getId()).getReminders();
+                List<Long> expectedIds = extractFutureFromTargetOrderedByRemindDateIds(bomReminders, target, count);
+
+                assertThat(foundReminders).extracting("id").containsExactlyElementsOf(expectedIds);
+            }
+        }
+
+        @DisplayName("messageId와 memberId로 단건 조회한다")
+        @Test
+        void findOne() {
+            Reminder target = bomReminders.get(5);
+            ReminderResponse response = reminderService.findOne(target.getMessage().getId(),
+                    target.getMember().getId());
+
+            assertThat(target.getId()).isEqualTo(response.getId());
+        }
+
+        @DisplayName("messageId와 memberId가 일치하는 리마인더가 없다면 예외가 발생한다")
+        @Test
+        void findOneThrowsException() {
+            Member kkojae = members.save(KKOJAE.create());
+            Message target = noticeMessages.get(0);
+
+            assertThatThrownBy(() -> reminderService.findOne(target.getId(), kkojae.getId()))
+                    .isInstanceOf(ReminderNotFoundException.class);
+        }
+
+        private List<Message> createAndSaveMessages(final Channel channel, final Member member) {
+            List<Message> messagesInChannel = Arrays.stream(MessageFixtures.values())
+                    .map(messageFixture -> messageFixture.create(channel, member))
+                    .collect(Collectors.toList());
+
+            for (Message fixture : messagesInChannel) {
+                messages.save(fixture);
+            }
+
+            return messagesInChannel;
```

Comment:
```suggestion
            return Arrays.stream(MessageFixtures.values())
                    .map(messageFixture -> messageFixture.create(channel, member))
                    .map(message -> messages.save(message))
                    .collect(Collectors.toList());
```

map을 두 번 하는걸로 해봤는데 어떨까요??
좀 더 간단하게 표현하고 싶어서 수정해봤어요!
(하고나니 stream에서 너무 많은 일을 하는건가? 라는 생각도 드네용)

Reply:
저는 로직을 따로 두는게 더 명확해보이는 것 같아요😄